x-pack/platform/packages/shared/onechat/onechat-common/tools/tool_result.ts

```diff
@@ -34,10 +32,15 @@ export interface ResourceResult {
 }
 
 export interface TabularDataResult {
+  toolResultId: string;
```

Comment:
- We are using `snake_case` for our domain model (because it's surfaced in our APIs) -> rename to `tool_result_id`

- I wonder, don't we want to have it optional and on all result types?

Reply:
1. Fixed the casing.
2. I think so but maybe just add it, as we add support for other artifacts?